Paws4Me/Paws4Me/Data/LocalPetNameViewController.swift

```diff
@@ -56,65 +72,66 @@ class LocalPetViewController: UIViewController {
-      cell.textLabel?.text = pet.value(forKeyPath: "petName") as? String
-        guard let imagePetLocal = (pet.value(forKeyPath: "petImage") as? String) else { return UITableViewCell() }
-        cell.imageView?.loadImageFromURL(imageURL: imagePetLocal)
-      return cell
-    }
-      func tableView(_ tableView: UITableView,
-                     commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
-
-          guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
-            return
-          }
-
-          let managedContext = appDelegate.persistentContainer.viewContext
-
-        if editingStyle == .delete {
-            managedContext.delete(pets[indexPath.row])
-          do {
-            try managedContext.save()
-            tableView.reloadData()
-          } catch _ as NSError {
-              displayAlert(alertTitle: "Unable to delete)",
-                                 alertMessage: "There was a problem deleting this pet",
-                                 alertActionTitle: "Try again" ,
-                           alertDelegate: self, alertTriggered: .fatalLocalDatabaseAlert)
-          }
-        }
-      }
-  }
+}
+// MARK: - UITableViewDataSource
+   extension LocalPetViewController: UITableViewDataSource {
+
+     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+         guard let countPet = pets?.count else {return 0}
+         return countPet
+     }
+
+     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+       let pet = pets?[indexPath.row]
+         guard let cell = tableView.dequeueReusableCell(withIdentifier: "favouritePetCell",
+                                                        for: indexPath) as? FavouriteTableViewCell else {
+             return UITableViewCell()
+         }
+         cell.namePet = pet?.value(forKeyPath: "petName") as? String ?? ""
+         cell.imagePet = pet?.value(forKeyPath: "petImage") as? String ?? ""
+         cell.setNeedsLayout()
+       return cell
+     }
+       func tableView(_ tableView: UITableView,
+                      trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+           let deleteAction = UIContextualAction(style: .normal, title: "Delete",
+                                           handler: { (_: UIContextualAction, _: UIView, success: (Bool) -> Void) in
+               DispatchQueue.main.async {
+                   self.showDeleteWarning(for: indexPath)
+               }
+               success(true)
+           })
+
+           deleteAction.image = UIImage(named: "delete")
+           deleteAction.backgroundColor = UIColor.myAppPurple
+
+           return UISwipeActionsConfiguration(actions: [deleteAction])
+       }
+
+       func showDeleteWarning(for indexPath: IndexPath) {
+           let alert = UIAlertController(title: "Delete Pet",
```

Comment:
Let's see if we can create a function that can be put in an extension so that we can re-use this code? Think along the lines of receiving a `handler` 😄

Reply:
Done 👍🏻